OnCampusTweaks.user.js

```diff
@@ -14,37 +14,35 @@
 // Thus, we must check independently to see if the full page has loaded.
 var checkExist = setInterval(function() {
     // if the string "assignment-center" is in the url
-    if (window.location.href.indexOf("assignment-center") > -1) { // TODO: find better way of doing this
+    if (window.location.href.indexOf("assignment-center") > -1) {
         if (document.readyState === 'ready' || document.readyState === 'complete') {
             changeToWeekView();
             hideCompletedTasks();
-            clearInterval(checkExist);
+            clearInterval(checkExist); //Run only one time. not needed to run again as the website saves your preferences for that session
         }
     }
-}, 100); // check every 100ms
-
-// Changes to week view
+},100)//100 ms
 function changeToWeekView() {
     // if the string "assignment-center" is in the url
     if (window.location.href.indexOf("assignment-center") > -1) {
         document.getElementById("week-view").click();
         console.log("OnCampusTweaks: Changed to week view");
     }
-
 };
 
 // hides completed tasks automatically in the assignment center
+// Unfortunately upon marking a assignment as completed, it will not dissapear however I am not going to mess with this as its most likely by design of the website and some people could prefer to still see it.
```

Comment:
I usually solve this by quickly refreshing the page, though I appreciate this being documented.